packages/x-date-pickers/src/hooks/useFieldPlaceholder.ts

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
```

Comment:
@LukasTy This was the main missing piece of API for this recipe.

Do you think it's worth publishing or should we let the user handle it? (by default they receive `format="L"` so they need to expand it which is not trivial without our tools).

We could also use it in the button and the autocomplete demo (which I plan to improve once we have this recipe and maybe the masked one).

Reply:
Just in case, we can't re-use it internally because the sections are deep inside state logic of the field.
I'm re-using all the internals, but this hook is only used in recipes and will probably always be.